storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -540,12 +540,16 @@
         topology (read-storm-topology-as-nimbus storm-id blob-store)
         executor->component (->> (compute-executor->component nimbus storm-id)
                                  (map-key (fn [[start-task end-task]]
-                                            (ExecutorDetails. (int start-task) (int end-task)))))]
+                                            (ExecutorDetails. (int start-task) (int end-task)))))
+        launch-time-secs (if storm-base (:launch-time-secs storm-base)
+                           (throw
+                             (NotAliveException. (str storm-id))))]
```

Comment:
We can throw the NotAliveException as soon as we know `storm-base` is `nil`, something like:

```Clojure
(defn read-topology-details [nimbus storm-id]
  (let [blob-store (:blob-store nimbus)
        storm-base (or
                     (.storm-base (:storm-cluster-state nimbus) storm-id nil)
                     (throw (NotAliveException. storm-id)))
        topology-conf (read-storm-conf-as-nimbus storm-id blob-store)
        topology (read-storm-topology-as-nimbus storm-id blob-store)
        executor->component (->> (compute-executor->component nimbus storm-id)
                                 (map-key (fn [[start-task end-task]]
                                            (ExecutorDetails. (int start-task) (int end-task)))))]
    (TopologyDetails. storm-id
                      topology-conf
                      topology
                      (:num-workers storm-base)
                      executor->component
                      (:launch-time-secs storm-base))))
```

This way we shortcut out of doing the extra work.

The snippet above also removes the unused binding for `conf`.